pkg/modulewriter/tfwriter.go

```diff
@@ -365,8 +344,10 @@ func getUsedDeploymentVars(group config.Group, bp config.Blueprint) map[string]c
 	for _, m := range group.Modules {
 		used = append(used, config.GetUsedDeploymentVars(m.Settings.AsObject())...)
 	}
-	for _, p := range getProviders(bp) {
-		used = append(used, config.GetUsedDeploymentVars(p.config.AsObject())...)
+	for _, v := range group.TerraformProviders {
+		used = append(used, config.GetUsedDeploymentVars(v.Configuration.AsObject())...)
+		used = append(used, config.GetUsedDeploymentVars(cty.StringVal(v.Source))...)
```

Comment:
Is your intent to support expressions in `source` and `version`? If yes, they can't be of type `string`. Otherwise don't make any changes here.